packages/leather-provider/src/index.ts

```diff
@@ -0,0 +1,189 @@
+import type { StacksProvider } from '@stacks/connect';
+
+import { type LeatherRpcMethodMap, type RpcParameter, RpcRequests } from '@leather.io/rpc';
+
+import { addLeatherToProviders } from './add-leather-to-providers';
+import type { WalletMethodNames, WalletResponses } from './rpc-methods';
+
+// TODO: change this to normal env variables
+const BRANCH = 'TEST_BRANCH';
+const COMMIT_SHA = 'TEST_COMMIT_SHA';
+const VERSION = 'VERSION';
+
+function isRN(): boolean {
+  return !!(window as any)?.ReactNativeWebView;
+}
+
+enum DomEventName {
+  request = 'request',
+}
+
+addLeatherToProviders();
+
+interface LeatherProviderOverrides extends StacksProvider {
+  isLeather: true;
+}
+
+const provider: LeatherProviderOverrides = {
+  isLeather: true,
+
+  getURL: () => {
+    // TODO: deprecated
+    throw new Error('This function is deprecated');
+  },
+
+  structuredDataSignatureRequest: () => {
+    // TODO: deprecated
+    throw new Error('This function is deprecated');
+  },
+
+  signatureRequest: () => {
+    // TODO: deprecated
+    throw new Error('This function is deprecated');
+  },
+
+  authenticationRequest: () => {
+    // TODO: deprecated
+    throw new Error('This function is deprecated');
+  },
+
+  transactionRequest: () => {
+    // TODO: deprecated
+    throw new Error('This function is deprecated');
+  },
+  psbtRequest: () => {
+    // TODO: deprecated
+    throw new Error('This function is deprecated');
+  },
+  profileUpdateRequest: () => {
+    // TODO: deprecated
+    throw new Error('This function is deprecated');
+  },
+
+  getProductInfo() {
+    return {
+      version: VERSION,
+      name: 'Leather',
+      meta: {
+        tag: BRANCH,
+        commit: COMMIT_SHA,
+      },
+    };
+  },
+
+  request(
+    method: WalletMethodNames,
+    params?: RpcParameter
+  ): Promise<LeatherRpcMethodMap[WalletMethodNames]['response']> {
+    const id: string = crypto.randomUUID();
+    const rpcRequest: RpcRequests = {
+      jsonrpc: '2.0',
+      id,
+      method,
+      params: (params ?? {}) as any,
+    };
+    if (isRN()) {
+      (window as any).ReactNativeWebView.postMessage(JSON.stringify(rpcRequest)); // RN
+    } else {
+      document.dispatchEvent(new CustomEvent(DomEventName.request, { detail: rpcRequest }));
+    }
```

Comment:
Can we pass in this as a callback, rather than having ternary logic? What if we add a third platform, do we keep extending? 

IMO here we should call `dispatchCallback(rpcRequest)` and pass in to a method to dispatch on each respective platform